app/src/main/java/bg/zahov/app/data/interfaces/WorkoutProvider.kt

```diff
@@ -13,6 +14,9 @@ interface WorkoutProvider {
     suspend fun getPastWorkouts(): Flow<List<Workout>>
     suspend fun addTemplateWorkout(newWorkout: Workout)
     suspend fun getTemplateExercises(): Flow<List<Exercise>>
+    suspend fun getExerciseByName(name: String): Flow<Exercise?>
+    suspend fun getExercisesByWrapper(exercises:List<ExerciseData>): Flow<List<Exercise>>
```

Comment:
Try to define another function name wrapper is too generic and it doesn't give any kind of context

Reply:
fixed